src/components/LicensesFieldArray/AmendmentsFieldArray.js

```diff
@@ -131,14 +126,42 @@ class AmendmentsFieldArray extends React.Component {
               <Row>
                 <Col xs={12} md={4}>
                   <Field
-                    component={Select}
-                    dataOptions={amendmentStatusValues}
-                    label={<FormattedMessage id="ui-agreements.license.prop.status" />}
                     name={`${name}[${i}].status`}
-                    placeholder=" "
-                    required
                     validate={validators.required}
-                  />
+                    label={<FormattedMessage id="ui-agreements.license.prop.status" />}
+                  >
+                    {props => (
+                      <Select
+                        {...props}
+                        dataOptions={amendmentStatusValues}
+                        onChange={(e) => {
+                          const { value } = e.target;
+                          let warning;
+                          let warningValuesStatus;
+                          if (value === statuses.CURRENT) {
+                            if (new Date(amendment.startDate).getTime() > new Date().getTime()) {
+                              warning = 'ui-agreements.license.warn.amendmentFuture';
+                            }
+                            // Amendment end date is in the past
+                            if (new Date(amendment.endDate).getTime() < new Date().getTime()) {
+                              warning = 'ui-agreements.license.warn.amendmentPast';
+                            }
+                            // Amendment has an invalid status.
+                            const linkedStatus = get(amendment, 'status', {});
+                            if (linkedStatus.value === statuses.EXPIRED || linkedStatus.value === statuses.REJECTED) {
+                              warning = 'ui-agreements.license.warn.amendmentStatus';
+                              warningValuesStatus = linkedStatus.label;
+                            }
+                          }
+                          statusWarnings[i].warningId = warning;
```

Comment:
This isn't how you set state for a component. See [the docs](https://reactjs.org/docs/react-component.html#setstate) or [this CSS tricks article](https://css-tricks.com/understanding-react-setstate/) (since the docs are down while I'm writing this).

Reply:
Yeah, I'm kinda surprised this worked looking back... code has been changed to remedy this.